### PR TITLE
Hook the Nix styling scripts into CI

### PR DESCRIPTION
**Pull request by monacoremo**
This should be faster, more consistent (as the same scripts can be used for local development) and hopefully less random than the stack-based integration :-)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -88,39 +88,18 @@ jobs:
...
-          name: Add stack tools to $PATH
-          command: |
-            echo "export PATH=/home/circleci/.local/bin:$PATH" >> $BASH_ENV
+          name: Install linting and styling scripts
+          command: nix-env -f default.nix -iA lint style
       - run:
-          name: run linter
-          command: make lint
+          name: Run linter
+          command: postgrest-lint
```
Comment (steve-chavez) [inline .circleci/config.yml:99]: For non-nix users, do you think we could suggest here a way of checking the failure locally?

I'm thinking of having an `echo` like:

```suggestion
          name: Run linter
          command: |
            echo "Note: For checking this locally, use `make lint`. Or, if using `nix`, run `nix-shell --run postgrest-lint`.
            postgrest-lint
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -88,39 +88,18 @@ jobs:
...
       - run:
-          name: run styler
-          command: make style
+          name: Run style check
+          command: postgrest-style-check
```
Comment (steve-chavez) [inline .circleci/config.yml:102]: Similarly here:

```suggestion
          name: Run style check
          command: |
            echo 'Note: For checking this locally, use `make style`. Or, if using `nix`, run `nix-shell --run postgrest-style`'
            postgrest-style
```

Comment (monacoremo) [reply .circleci/config.yml]: Yes, that makes a lot of sense! Good idea :-)

File: .circleci/config.yml [hunk 3]
```diff
@@ -88,39 +88,22 @@ jobs:
...
           command: |
-            echo "export PATH=/home/circleci/.local/bin:$PATH" >> $BASH_ENV
-      - run:
-          name: run linter
-          command: make lint
-      - run:
-          name: run styler
-          command: make style
+            echo 'Note: For checking this locally, use `make style`. Or, if using `nix`, run `nix-shell --run postgrest-style`'
+            postgrest-style
```
Comment (steve-chavez) [inline .circleci/config.yml:106]: Made a mistake in my suggestion here. It should have been:

```suggestion
            postgrest-style-check
```

**Pull request merged**